src/core/libmaven/PeakDetector.cpp

```diff
@@ -304,77 +304,78 @@ void PeakDetector::alignSamples(const int& method) {
     }
 }
 
-void PeakDetector::processSlices(vector<mzSlice *> &slices, string setName)
+void PeakDetector::processSlices(vector<mzSlice*> &slices, string setName)
 {
-
-    if (slices.size() == 0)
+    if (slices.empty())
         return;
-    mavenParameters->allgroups.clear();
 
+    mavenParameters->allgroups.clear();
     sort(slices.begin(), slices.end(), mzSlice::compIntensity);
-
-    int eicCount = 0;
-    for (unsigned int s = 0; s < slices.size(); s++)
-    {
-
+    for (unsigned int s = 0; s < slices.size(); s++) {
         if (mavenParameters->stop)
             break;
-        mzSlice *slice = slices[s];
 
-        Compound *compound = slice->compound;
-
-        if (compound != NULL && compound->hasGroup())
+        mzSlice* slice = slices[s];
+        Compound* compound = slice->compound;
+        if (compound != nullptr && compound->hasGroup())
             compound->unlinkGroup();
 
-        vector<EIC *> eics = pullEICs(slice,
-                                      mavenParameters->samples,
-                                      mavenParameters);
+        vector<EIC*> eics = pullEICs(slice,
+                                     mavenParameters->samples,
+                                     mavenParameters);
 
         if (mavenParameters->clsf->hasModel())
-        {
             mavenParameters->clsf->scoreEICs(eics);
+        auto highestIntensity = 0.0f;
+        auto mzAtHighestIntensity = 0.0f;
+        auto rtAtHighestIntensity = 0.0f;
+
+        for (auto eic: eics) {
+            for (size_t i = 0; i < eic->size(); ++i) {
+                auto intensityAtIdx = eic->intensity[i];
+                if (intensityAtIdx > highestIntensity) {
+                    highestIntensity = intensityAtIdx;
```

Comment:
@saifulbkhan are these values being used somewhere? can't find it.

Reply:
Sorry, these were left behind. Was using them to export slices as CSV. Will get rid of it.